Repositories/DbRepository.cs

```diff
@@ -77,9 +77,14 @@ public async Task<TEntity> GetByIdAsync(int id)
             return await entities.FindAsync(id);
         }
 
-        public void Update(TEntity id)
+        public void Update(TEntity entity)
         {
-            _context.SaveChanges();
+           var DbEntity = entities.Find(entity.Id);
+           if(DbEntity != null)
+           {
+                _context.Entry(DbEntity).CurrentValues.SetValues(entity);
+                _context.SaveChanges();
+           }
```

Comment:
I think there is a method entities.Update(entity) that could work here.

Reply:
Updated the method.